build.py

```diff
@@ -34,37 +62,37 @@ def print_usage():
             name += " (default with --debug)"
         configs.append(name + "\n    " + " ".join(args))
     configs_string = "\n  ".join(configs)
-    cmake_name = os.path.basename(CMAKE)
-    generator_name = CMAKE_GENERATOR.lower()
-    default_config_name = DEFAULT_CONFIG_NAME
-    debug_config_name = DEBUG_CONFIG_NAME
-    print(f"""Usage: {script_name} [BUILD [BUILD ...]] [--all] [--debug] [MAKE_OPTIONS]
-
-Build one or more predefined build configurations of Fast Downward. Each build
-uses {cmake_name} to compile the code using {generator_name} . Build configurations
-differ in the parameters they pass to {cmake_name}. By default, the build uses all
-available cores if this number can be determined. Use the "-j" option for
-{cmake_name} to override this default behaviour.
-
-Build configurations
+    epilog = f"""build configurations:
   {configs_string}
 
---all         Alias to build all build configurations.
---debug       Alias to build the default debug build configuration.
---help        Print this message and exit.
-
-Make options
-  All other parameters are forwarded to the build step.
-
-Example usage:
-  ./{script_name}                     # build {default_config_name} in #cores threads
-  ./{script_name} -j4                 # build {default_config_name} in 4 threads
-  ./{script_name} debug               # build debug
-  ./{script_name} --debug             # build {debug_config_name}
-  ./{script_name} release debug       # build release and debug configs
-  ./{script_name} --all VERBOSE=true  # build all build configs with detailed logs
-""")
-
+example usage:
+  {script_name}                     # build {DEFAULT_CONFIG_NAME} with #cores parallel processes
+  {script_name} -j4                 # build {DEFAULT_CONFIG_NAME} with 4 parallel processes
```

Comment:
Trying this command leads to "build.py: error: unrecognized arguments: -j4" for me; but `build.py release -j4` for example works.

Reply:
The problem is likely that build.argparse tries to recognize `-j4` as options rather than positional arguments (cf. the `--` discussion we recently had; I expect `./build.py -- -j4` would also work, but of course we want to make it easier than that). Using `argparse.parse_known_args` may or may not be good enough to address this.

Reply:
`./build.py -- -j4` doesn't give me an error but It also does not do anything (i.e. no output whatsoever).

Reply:
We fixed it using`argparse.parse_known_args` as suggested.